mlir/test/compiler-course/chistov_a_loop_trace_tests/chistov_a_test.mlir

```diff
@@ -0,0 +1,82 @@
+// RUN: mlir-opt -load-pass-plugin=%mlir_lib_dir/LoopTracePass_Chistov_Alexey_FIIT1_MLIR%shlibext \
+// RUN: --pass-pipeline="builtin.module(LoopTracePass)" %s | FileCheck %s
+
+module {
+  // CHECK: func.func private @trace_loop_iter_end()
+  // CHECK-NEXT: func.func private @trace_loop_iter_begin()
+  // CHECK-LABEL: func.func @affine_for()
+  func.func @affine_for() -> i32 {
+    %sum_init = arith.constant 0 : i32
+    %result = affine.for %i = 0 to 10 iter_args(%sum_iter = %sum_init) -> i32 {
+      // CHECK: func.call @trace_loop_iter_begin() : () -> ()
+      %i_i32 = arith.index_cast %i : index to i32
+      %new_sum = arith.addi %sum_iter, %i_i32 : i32
+      // CHECK: func.call @trace_loop_iter_end() : () -> ()
+      affine.yield %new_sum : i32
+    }
+    return %result : i32
+  }
+
+  // CHECK-LABEL: func.func @scf_for()
+  func.func @scf_for() -> i32 {
+    %sum_init = arith.constant 0 : i32
+    %lower = arith.constant 0 : index
+    %upper = arith.constant 10 : index
+    %step = arith.constant 1 : index
+
+    %result = scf.for %i = %lower to %upper step %step iter_args(%sum_iter = %sum_init) -> i32 {
+      // CHECK: func.call @trace_loop_iter_begin() : () -> ()
+      %i_i32 = arith.index_cast %i : index to i32
+      %new_sum = arith.addi %sum_iter, %i_i32 : i32
+      // CHECK: func.call @trace_loop_iter_end() : () -> ()
+      scf.yield %new_sum : i32
+    }
+
+    return %result : i32
+  }
+
+  // CHECK-LABEL: func.func @scf_while()
+  func.func @scf_while() -> i32 {
+    %sum_init = arith.constant 0 : i32
+    %i_init = arith.constant 0 : i32
+    %limit = arith.constant 10 : i32
+
+    %result:2 = scf.while (%sum = %sum_init, %i = %i_init) : (i32, i32) -> (i32, i32) {
+      %cmp = arith.cmpi slt, %i, %limit : i32
+      scf.condition(%cmp) %sum, %i : i32, i32
+    } do {
+    ^bb0(%sum_arg: i32, %i_arg: i32):
+      // CHECK: func.call @trace_loop_iter_begin() : () -> ()
```

Comment:
Please, check that insertion position is correct. Current test only checks the presense of the trace markers in the function